vue_frontend/src/main.js

```diff
@@ -2,5 +2,8 @@ import { createApp } from 'vue'
 import App from './App.vue'
 import router from './router'
 import store from './store'
+import axios from 'axios'
 
-createApp(App).use(store).use(router).mount('#app')
+axios.defaults.baseURL = 'http://127.0.0.1:8000'
+
+createApp(App).use(store).use(router, axios).mount('#app')
```

Comment:
I don't think this is wrong but Axios allows you  to create an Axios instance like this:

```js
const api = axios.create({ baseURL: 'http://127.0.0.1:8000' })

createApp(App).use(store).use(router, axios).mount('#app')
```

Is not a good practice to define your API URL like this. You can create a `.env` file on the `vue_frontend` directory and define your API URL over there. It will allow other devs to change the API ULR without having to change the code. And it will allow you to change the URL when you deploy it to production.

Like this:
`/vue_frontend/.env`:
```env
BASE_URL=http://localhost:8000
# or
VUE_APP_API_URLhttp://localhost:8000
```

`/vue_frontend/src/main.js`:
```js
const api = axios.create({ baseURL: process.env.VUE_APP_API_URL })
// or
// const api = axios.create({ baseURL: process.env.BASE_URL })

createApp(App).use(store).use(router, axios).mount('#app')
```

Read more on: https://cli.vuejs.org/guide/mode-and-env.html#modes

Reply:
Another good practice is create a file with all the routes that you will use in order to centralize and hide from the the application details related to API.

You can create a function to use the API routes that lead with `messages` like this:
```js
// ./vue_frontend/src/api/messages.js
const api = axios.create({ baseURL: process.env.BASE_URL })

const sendMessage = async (message) => {
    return await api.post('/messsages', message);
}

const getMessages = async () => {
    return await api.get('/messages');
}

export { getMessages };
```

 Split the code in files, like `user.js`, `projects.js`, and so on. 
This way in the future you can change details and, if you guarantee to deliver the same return, without need to change the code that already uses this functions.